martian-lab/examples/sum_sq/src/main.rs

```diff
@@ -40,9 +40,14 @@ fn main() -> Result<(), Error> {
 
     if args.cmd_martian {
         // Call the martian adapter
-        martian_main(args.arg_adapter, stage_registry)?;
-    // If you want explicit control over the log level, use:
-    // martian_main_with_log_level(...)
+        let adapter = MartianAdapter::new(stage_registry);
+
+        // If you want explicit control over the log level use for example:
+        // let adapter = adapter.log_level(LevelFilter::Info);
+
+        let retcode = adapter.run(args.arg_adapter);
+        std::process::exit(retcode);
```

Comment:
@pmarks  Why do we want to explicitly exit here? Doesn't a `main()` that return a `Result` have an exit code of `0` for `Result::Ok` and an exit code of `1` for `Result::Err`?

Reply:
Well once the error reporting is fixed to exit without returning to here, yes, but in the current form falling through here would yield `Ok`.